cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+  const options = processConfig(cmdArgs);
+  interrupt.on_interrupt(() => teardown());
+  return Promise.resolve().bind({ options })
+    .then(setup)
+    .then(validateMigration)
+    .then(makeBackup)
+    .then(renameUserTables)
+    .then(moveInternalTables)
+    .then(renameIndices)
+    .then(rewriteHzCollectionDocs)
+    .then(exportNewSchema)
+    .then(teardown)
+    .catch((err) => {
+      red(` └── ${err.message}`);
+      return Promise.bind(options).then(teardown());
+    });
+}
+
+function green(arg, ...args) {
+  console.log(chalk.green(arg), ...args);
+}
+
+function red(arg, ...args) {
+  console.error(chalk.red.bold(arg), ...args);
+}
+
+function white(arg, ...args) {
+  console.log(chalk.white(arg), ...args);
+}
+
+function processConfig(cmdArgs) {
+  // do boilerplate to get config args :/
+  const parser = new argparse.ArgumentParser({ prog: 'hz migrate' });
+
+  parser.addArgument([ 'project_path' ], {
+    default: '.',
+    nargs: '?',
+    help: 'Change to this directory before migrating',
+  });
+
+  parser.addArgument([ '--project-name', '-n' ], {
+    help: 'Name of the Horizon project server',
+  });
+
+  parser.addArgument([ '--connect', '-c' ], {
+    metavar: 'host:port',
+    default: undefined,
+    help: 'Host and port of the RethinkDB server to connect to.',
+  });
+
+  parser.addArgument([ '--rdb-user' ], {
+    default: 'admin',
+    metavar: 'USER',
+    help: 'RethinkDB User',
+  });
+
+  parser.addArgument([ '--rdb-password' ], {
+    default: undefined,
+    metavar: 'PASSWORD',
+    help: 'RethinkDB Password',
+  });
+
+  parser.addArgument([ '--start-rethinkdb' ], {
+    action: 'storeTrue',
+    help: 'Start up a RethinkDB server in the current directory'
+  });
+
+  parser.addArgument([ '--config' ], {
+    default: '.hz/config.toml',
+    help: 'Path to the config file to use, defaults to ".hz/config.toml".',
+  });
+
+  parser.addArgument([ '--skip-backup' ], {
+    action: 'storeTrue',
+    help: 'Whether to perform a backup of rethinkdb_data before migrating',
+  });
+
+  const parsed = parser.parseArgs(cmdArgs);
+  const confOptions = config.read_from_config_file(parsed.project_path, parsed.config);
+  // Pull out the relevant settings from the config file
+  const options = {
+    project_path: parsed.project_path || '.',
+    project_name: parsed.project_name || confOptions.project_name,
+    rdb_host: parsed.rdb_host || confOptions.rdb_host || 'localhost',
+    rdb_port: parsed.rdb_port || confOptions.rdb_port || 28015,
+    rdb_user: parsed.rdb_user || confOptions.rdb_user || 'admin',
+    rdb_password: parsed.rdb_password || confOptions.rdb_password || '',
+    skip_backup: parsed.skip_backup || false,
+    start_rethinkdb: parsed.start_rethinkdb || false,
+  };
+  // sets rdb_host and rdb_port from connect if necessary
+  if (parsed.connect) {
+    config.parse_connect(parsed.connect, options);
+  }
+
+  if (options.project_name == null) {
+    throw new Error('project_name is null');
```

Comment:
This error message seems a little misleading - I imagine this indicates that `project_name` has not been set, may as well say that.